File: 1.cs
using System;
class Program
{
    static string AddSegment(string main, string domain, int ibegin, int iend)
    {
        if (main[ibegin - 1] == '1')
            domain += main[ibegin - 1];
        
        for (int i = ibegin; i <= iend; i++)
            domain += main[i];
        
        return domain + ", ";
    }

    static void Main()
    {
        Console.Write("Введите строку: ");
        string str1 = Console.ReadLine();
        string str2 = "";

        for (int i = 1; i < str1.Length; i++)
        {
            if (str1[i] == '0' && str1[i - 1] == '1')
            {
                int iend = i;
                while (iend < str1.Length && str1[iend] == '0')
                    iend++;
                
                if (iend < str1.Length && str1[iend] == '1')
                    str2 = AddSegment(str1, str2, i - 1, iend);
            }
        }

        if (str2.Length > 0)
            str2 = str2.Remove(str2.Length - 2, 1);
        
        Console.WriteLine(str2);
    }
}

File: 1.swift
func addSegment(main: String, domain: String, ibegin: Int, iend: Int) -> String {
    var newDomain = domain
    let mainChars = Array(main)
    
    if ibegin > 0 && mainChars[ibegin - 1] == "1" {
        newDomain.append(mainChars[ibegin - 1])
    }
    
    for i in ibegin...iend {
        if i < mainChars.count {
            newDomain.append(mainChars[i])
        }
    }
    
    return newDomain + ", "
}

func main() {
    print("Введите строку: ", terminator: "")
    guard let str1 = readLine() else { return }
    
    var str2 = ""
    let chars = Array(str1)
    
    for i in 1..<chars.count {
        if chars[i] == "0" && chars[i-1] == "1" {
            var iend = i
            while iend < chars.count && chars[iend] == "0" {
                iend += 1
            }
            
            if iend < chars.count && chars[iend] == "1" {
                str2 = addSegment(main: str1, domain: str2, ibegin: i-1, iend: iend)
            }
        }
    }
    
    if !str2.isEmpty {
        str2.removeLast(2)
    }
    
    print(str2)
}

main()
